pypsa/optimization/constraints.py

```diff
@@ -624,6 +624,42 @@ def define_fixed_nominal_constraints(n, c, attr):
     n.model.add_constraints(var, "=", fix, f"{c}-{attr}_set")
 
 
+def define_modular_constraints(n, c, attr, attr_nom):
+    """
+    Sets constraints for fixing modular variables of a given component. It
+    allows to define optimal capacity of a component as multiple of the nominal
+    capacity of the single module.
+
```

Comment:
Instead of overloading "p/e_nom" let's define a new parameter "p/e_nom_mod" that is used to identify the size of the modular component.
That requires little changes in the code below and the addition of new parameters to be loaded during the add(...) function